CHANGES.md

```diff
@@ -5,8 +5,24 @@
 * Adjust classes to abide by naming conventions for attributes.
 * Change `ExponentiatedGradient` signature by renaming argument `T` to
   `max_iter`, `eta_mul` to `eta0`, and by adding `run_linprog_step`.
+* API refactoring to separate out different uses of `eps` within
+  `ExponentiatedGradient`. It is now solely responsible for setting a bound on
+  the lambda vectors. The other usage of `eps` as a bound for constraints is
+  now captured directly on the moment classes as follows:
+  * Classification moments: `ConditionalSelectionRate` and its subclasses
+    have new arguments on the constructor:
+    * `difference_bound` - for difference-based constraints such as demographic
+      parity difference
+    * `ratio_bound_slack` - for ratio-based constraints such as demographic
+      parity ratio
+    * Additionally, there's a `ratio_bound` argument which represents the
+      argument previously called `ratio`.
+  * Regression moments: `ConditionalLossMoment` and its subclasses have a new
+    argument `upper_bound` with the same purpose for newly enabled regression
+    scenarios on `ExponentiatedGradient`.
```

Comment:
@romanlutz I think this in intentionally developer-facing, but I'd vote to make this more user-facing.  If someone doesn't already know how the guts of the library works, there's nothing in here that communicates "the way you set bounds on DemographicParity has changed."  Relatedly, has the behavior of `DemographicParity()` changed?

Reply:
This came up on another PR already.

CHANGES.md is only shown in the repo and intentionally intended for developers or folks who are familiar with the code. We should have a section on the webpage (that we link from the README perhaps) to say how to upgrade from version X to Y (for all versions). 

I'm not going to address this in this PR, though, because it'll already get pretty huge with the user guide updates I'm working on right now.